functions/security/security_system_test.go

```diff
@@ -16,22 +16,121 @@ package security
 
 import (
 	"bytes"
+	"fmt"
+	"log"
 	"os"
+	"os/exec"
+	"strings"
 	"testing"
+	"time"
 )
 
+const RuntimeVersion = "go113"
+
+func TestMain(m *testing.M) {
+	// Only run end-to-end tests when configured to do so.
+	if os.Getenv("GOLANG_SAMPLES_E2E_TEST") == "" {
+		log.Println("Skipping end-to-end tests: GOLANG_SAMPLES_E2E_TEST not set")
+		os.Exit(m.Run())
+	}
+
+	retn, err := setupAndRun(m)
+	if err != nil {
+		log.Fatal(err)
+	}
+	os.Exit(retn)
+}
+
+func setupAndRun(m *testing.M) (int, error) {
+	entryPoint := "MakeGetRequestFixture"
+	targetName := entryPoint + "-" + time.Now().Format("20060102-150405")
+
+	// Setup function for tests.
+	cmd := exec.Command("gcloud", "functions", "deploy", targetName,
+		"--entry-point="+entryPoint,
+		"--runtime="+RuntimeVersion,
+		"--no-allow-unauthenticated",
+		"--trigger-http",
+	)
+	log.Printf("Running: %s %s", cmd.Path, strings.Join(cmd.Args, " "))
+	if _, err := cmd.Output(); err != nil {
+		log.Println(string(err.(*exec.ExitError).Stderr))
+		return 1, fmt.Errorf("Setup: Deploy target function: %w", err)
+	}
+
+	// Tear down the deployed function.
+	defer func() {
+		cmd = exec.Command("gcloud", "functions", "delete", targetName)
+		log.Printf("Running: %s %s", cmd.Path, strings.Join(cmd.Args, " "))
+		if _, err := cmd.Output(); err != nil {
+			log.Println(string(err.(*exec.ExitError).Stderr))
+			log.Printf("Teardown: Delete target function: %v", err)
+		}
+	}()
+
+	// Setup Relay service where the authenticated request will be run.
+	cmd = exec.Command("gcloud", "functions", "describe", targetName, "--format=value(httpsTrigger.url)")
+	log.Printf("Running: %s %s", cmd.Path, strings.Join(cmd.Args, " "))
+	out, err := cmd.Output()
+	if err != nil {
+		log.Println(string(err.(*exec.ExitError).Stderr))
+		return 1, fmt.Errorf("Setup: Get target function URL: %w", err)
+	}
+	targetURL := strings.TrimSpace(string(out))
+	relayName := entryPoint + "-relay-" + time.Now().Format("20060102-150405")
+
+	// Setup function for tests.
+	cmd = exec.Command("gcloud", "functions", "deploy", relayName,
+		"--entry-point="+entryPoint,
+		"--runtime="+RuntimeVersion,
+		"--no-allow-unauthenticated",
+		"--update-env-vars", "TARGET="+targetURL,
```

Comment:
Should this be `TARGET_URL`?

Reply:
This test is deploying two Cloud Functions. This one is deploying the Cloud Function which will be tested.

Toward the end of setupAndRun() os.Setenv is called to set `BASE_URL` for the test. Otherwise this function doesn't use env vars for URLs.